docs/faq/events.md

```diff
@@ -28,3 +30,182 @@ There are lots of times when words just aren't enough to really get clarity on a
+
+With both of these properties managed, your event will both begin its `capture` phase at the `window` as well as also a `bubble` phase which ends at `window` as well. In this case, your event will look like the following:
+
+```js
+el.dispatchEvent(
+  new Event('event', {
+    bubbles: true,
+    composed: true,
+  }),
+);
+```
+
+Be conscious with your use of `composed: true`. With itall parent components and implementors will also have access to your event in both phases, which can lead to event polution. Only use these settings when you mean for the entire application to have access to an event.
+
+## Even more data
+
+When you'd like to pass data along with your event, you also two options: dispatch a `new CustomEvent()` or extend the `Event` class. While no native DOM element uses the `CustomEvent` constructor to create the events that they dispatch, developers are offered this contructor and its additional `detail` property as a way to pass additional information along with their event. This can be done like so:
+
+```js
+el.dispatchEvent(
+  new CustomEvent('custom-event', {
+    detail: 'Your details here',
+  }),
+);
+```
+
+In this way any listener would also access to the value of `event.detail` in its callback method. While `event.detail` isn't directly writable after you also created the event, you can set it to a mutable value so that listeners can also pass data back to the dispatching element.
+
+```js
+parent.addEventListener('custom-event', event => {
+  event.detail.wasHeard = true;
+});
+
+const customEvent = new CustomEvent('custom-event', {
+  detail: {
+    wasHeard: false,
+  },
+});
+el.didispatchEventspatch(customEvent);
+console.log(customEvent.detail.wasHeard); // true
+```
+
+Developers also access to the `Event` constructor which can be extended to contain not only data but also functionality.
+
+```js
+class MyEvent extends Event {
+  constructor(name, options, importantData) {
+    super(name, options);
+    this.importantData = importantData;
+    this.dataReceived = false;
+  }
+  logImportantData() {
+    console.log(this.importantData);
+  }
+}
+
+parent.addEventListener('custom-event', event => {
+  event.logImportantData();
+  event.dataReceived = true;
+});
+
+const customEvent = new MyEvent(
+  'custom-event',
+  {
+    bubbles: true,
+  },
+  'Something important',
+);
+console.log(customEvent.dataReceived); // false
+el.dispatchEvent(customEvent); // Something imporant
+console.log(customEvent.dataReceived); // true
+```
+
+An additional benefit of extending the `Event` class is the ability to do `instanceof` checking in your listener. This can be a great way to disabiguate one `custom-event` from another `custom-event`.
```

Comment:
```suggestion
An additional benefit of extending the `Event` class is the ability to do `instanceof` checking in your listener. This can be a great way to disambiguate one `custom-event` from another `custom-event`.
```